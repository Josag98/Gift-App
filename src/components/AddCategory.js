import React,{ useState } from 'react'
import PropTypes from "prop-types";


export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('')

    const handleOnChange =(e)=>{
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) =>
    {
        e.preventDefault();

        if(inputValue.trim().length>2){
            setCategories(cat=>[inputValue,...cat]);
            setinputValue('')
        }
        

    }
    

  return (
      <form onSubmit={handleSubmit}>
    <input
    placeholder='Â¿Que gift desea buscar?'
    type="text"
    value={inputValue}
    onChange={handleOnChange}
    >
        
    </input>
    </form>
  )
}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}
