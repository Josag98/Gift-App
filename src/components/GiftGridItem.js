import React from 'react'
import PropTypes from "prop-types";

export const GiftGridItem =({title,url}) => {
  
  return (
    <div className='card animate__animated animate__swing'>
        <img src={url} alt={title}></img>
        <p>{title}</p>
    </div>
  )
}

GiftGridItem.propTypes={
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}
