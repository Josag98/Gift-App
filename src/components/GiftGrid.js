import React, { useEffect,useState } from 'react'
import { getGifts } from '../helpers/getGifts';
import {GiftGridItem} from './GiftGridItem';

export const GiftGrid = ({category}) => {
  const [images, setImages] = useState([]);

  

  useEffect(() => {
    getGifts(category).then(imgs => setImages(imgs))
  }, [category]);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => {
          return <GiftGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
}
