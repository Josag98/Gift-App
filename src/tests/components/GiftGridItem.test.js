import React from 'react';
import {shallow} from "enzyme"
import "@testing-library/jest-dom";
import { GiftGridItem } from '../../components/GiftGridItem';

describe('Pruebas del componentes GiftGridItem.js', () => {
  const url = "Esto es una url";
  const title = "Esto es un titulo";

  const wrapper = shallow(<GiftGridItem url={url} title={title} />);

  test("Se debe de renderizar GiftGridItem correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("El prop title debe de estar dentro de un parrafo", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("La url y el title de la imagen deben de ser los de los props", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test('El div debe de tener la clase animate__swing', () => { 

    const div = wrapper.find('div');
    const className = div.prop('className');

    expect(className.includes('animate__swing')).toBe(true);
   })
})