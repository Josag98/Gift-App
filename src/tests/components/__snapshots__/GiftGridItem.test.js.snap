// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componentes GiftGridItem.js Se debe de renderizar GiftGridItem correctamente 1`] = `
<div
  className="card animate__animated animate__swing"
>
  <img
    alt="Esto es un titulo"
    src="Esto es una url"
  />
  <p>
    Esto es un titulo
  </p>
</div>
`;
